### DISCUSSION TITLE
ClamAV antivirus configuration is not working correctly

### DISCUSSION THREAD

**Discussion by comsyspro** [LiveHelperChat/livehelperchat#1857] [General]
when i activate `Uploaded files from visitors are scanned` in `file configuration` and the visitor uploads a file everytime i get an error message virus `found in file!`

![image](https://user-images.githubusercontent.com/13809377/208412437-1faca216-fe0e-4149-9c16-704b207346ae.png)

in the clamav.log there is an error message **Can't open file or directory ERROR**:
```
tail -f /var/log/clamav/clamav.log
Mon Dec 19 11:06:09 2022 -> /var/www/html/lhc_web/var/storage/2022y/12/19/5/4d255ec19ef54ed1a9ac6f13f074a68c: Can't open file or directory ERROR
```
after checking this file path /var/www/html/lhc_web/var/storage/2022y/12/19/5/4d255ec19ef54ed1a9ac6f13f074a68c there is not such a directory or file. so there is somewhere a problem when uploading the file from the visitor site so that it isn't saved successfully. the storage folder stays empty in which normally should the uploaded file be placed. when uploading a file from admin's site the files are placed in the folder /var/www/html/lhc_web/var/storage/2022y/.

**Comment by remdex** [comment]
Make sure your clamv configuration is correct. Socket path etc in file configuration.

**Reply by comsyspro** [reply]
this should be right configured /etc/clamav/clamd.conf (only changed LocalSocketGroup and  User to root):
```
#Automatically Generated by clamav-daemon postinst
#To reconfigure clamd run #dpkg-reconfigure clamav-daemon
#Please read /usr/share/doc/clamav-daemon/README.Debian.gz for details
LocalSocket /var/run/clamav/clamd.ctl
FixStaleSocket true
LocalSocketGroup root
LocalSocketMode 666
# TemporaryDirectory is not set to its default /tmp here to make overriding
# the default with environment variables TMPDIR/TMP/TEMP possible
User root
```
the socket in `ClamAV socket path`  is /var/run/clamav/clamd.ctl like defined in the clamd.conf file. what is wrong?

**Reply by remdex** [reply]
That you will have to figureout. If file upload without clamv activation works, means problem somewhere else on your side :)

**Reply by comsyspro** [reply]
yes without clamav upload works, or do i have to change `ClamAV socket length` which is set default to 20000? 

after running php cron.php -s site_admin -c cron/util/test_antivirus -p package.json ithere is this error:
```
Warning: socket_connect(): unable to connect [2]: No such file or directory in /var/www/html/lhc_web/lib/core/lhexternal/Clamav.php on line 50
File is infected
```

**Reply by comsyspro** [reply]
found problem. clamav was blocked by application armor. now it works fine. thank you.
